scripts/get-flattened-publication.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -ueo pipefail
+
+USAGE="$0 HOST STORAGE_KEY
+Returns flat JSON corresponding with chain storage data returned by an RPC node.
+
+  HOST is a URL prefix against which an abci_query HTTP request may be issued,
+       e.g. \"https://xnet.rpc.agoric.net/\".
+  STORAGE_KEY is a chain storage key name, e.g. \"published.amm.pool0.metrics\".
+";
+
+URL_PREFIX="${1:-}";
+STORAGE_KEY="${2:-}";
+
+if [ ":${1:-}" = '--help' -o ":$URL_PREFIX" = : -o ":$STORAGE_KEY" = : -o $# -gt 2 ]; then
+  printf '%s' "$USAGE"
+  exit 64
+fi
+
+SCRIPTS_DIR="$(dirname "$(realpath "$0")")"
+FLATTENER="$SCRIPTS_DIR/flatten_json_sequence.awk"
+
+# Make the request and extract data from a response that looks like
+#   {
+#     "jsonrpc": "2.0",
+#     "id": -1,
+#     "result": {
+#       "response": {
+#         "code": 0,
+#         "log": "",
+#         "info": "",
+#         "index": "0",
+#         "key": null,
+#         "value": "<base64-encoded JSON text>",
+#         "proofOps": null,
+#         "height": "<decimal digits>",
+#         "codespace": ""
+#       }
+#     }
+#   }
+vars="$(
+  curl -sS "${URL_PREFIX%/}/abci_query?path=%22/custom/vstorage/data/$STORAGE_KEY%22" | \
```

Comment:
Incidentally, this is probably a bug somewhere in Cosmos/Tendermint. Their [documentation](https://docs.tendermint.com/master/rpc/#/ABCI/abci_query) gives example values of `path` like `/a/b/c`, but making a query with `?path=/a/b/c` yields a response like
```
{
  "jsonrpc": "2.0",
  "id": -1,
  "error": {
    "code": -32602,
    "message": "Invalid params",
    "data": "error converting http params to arguments: invalid character '/' looking for beginning of value"
  }
}
```
Something is apparently JSON-decoding the parameter value rather than using it directly.

Reply:
Specifically https://github.com/tendermint/tendermint/blob/75d51e18f740c7cbfb7d8b4d49182ee6c7f41982/rpc/jsonrpc/server/http_uri_handler.go#L42-L51 , but I can't really get to the bottom of the reflection-heavy `nonJSONStringToArg` in `httpParamsToArgs`.

Reply:
Opened https://github.com/tendermint/tendermint/issues/9164

Reply:
urlencode?

Reply:
No, I get the same results with `%22` vs. `"`. I even replaced an `a` with `\u0061` (which is a JSON escape sequence) in the quote-wrapped version and it still worked.